### PR TITLE
[New Model] QualityTaskAttachment 1.0.0

### PR DESCRIPTION
**Pull request by JulianSchererZF**
## Description

 This new model allows the quality use case to provide a detailed description of files attached to a quality task and transferred in a ZIP folder.

 --> 

New Model Request Issue: [174](https://github.com/eclipse-tractusx/sldt-semantic-models/issues/174)

Closes #

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the BAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar bamm-cli.jar -i \<path-to-aspect-model\> -v ). The  BAMM CLI is available [here](https://openmanufacturingplatform.github.io/sds-documentation/sds-developer-guide/dev-snapshot/tooling-guide/bamm-cli.html) and in [GitHub](https://github.com/OpenManufacturingPlatform/sds-sdk/releases)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "name" and "description"** in English language. 
- [x] **no duplicate names or preferredNames** within an Aspect (e.g. a Property and the referenced Characteristic should not have the same name)
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the BAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://openmanufacturingplatform.github.io/sds-documentation/bamm-specification/v1.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'

### CODE REVIEW

File: io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+:schema a samm:Property ;
+   samm:preferredName "schema"@en ;
+   samm:description "schema definition for file"@en ;
+   samm:characteristic :Schema .
+
+:filePath a samm:Property ;
+   samm:preferredName "File Path"@en ;
+   samm:description "Indicates where the file is located within the ZIP-folder"@en ;
+   samm:characteristic :Text ;
```
Comment (bs-jokri) [inline io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl:84]: something went south here. Replace with samm-c:Text also remove characteristics definition.

File: io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+   samm:description "Indicates which file format the file has"@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "csv" .
+
+:Schema a samm:Characteristic ;
+   samm:preferredName "Schema"@en ;
+   samm:description "Describes the schema of the values contained in the file"@en ;
+   samm:dataType :SchemaDefinition .
+
+:Text a samm:Characteristic ;
```
Comment (bs-jokri) [inline io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl:110]: this one needs to be removed. Probably the editor copied that into the model

File: io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+   samm:description "The data type of the data contained in the column"@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "double" .
+
+:unit a samm:Property ;
+   samm:preferredName "Unit"@en ;
+   samm:description "The unit of the data contained in the column (used for numeric values)"@en ;
+   samm:characteristic samm-c:UnitReference ;
+   samm:exampleValue "degreeCelsiusPerHour" ;
+   samm:see <https://docs.bosch-semantic-stack.com/samm-specification/2.0.0/appendix/unitcatalog.html> .
```
Comment (bs-jokri) [inline io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl:170]: replace with
https://eclipse-esmf.github.io/samm-specification/2.0.0/appendix/unitcatalog.html

**Review by bs-jokri — CHANGES_REQUESTED** [review]
please check comments

File: io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+   samm:description "Additional data for quality task"@en ;
+   samm:properties ( :qualityTaskId :parentAspectModel :files ) ;
+   samm:operations ( ) ;
+   samm:events ( ) .
+
+:qualityTaskId a samm:Property ;
+   samm:preferredName "Quality Task ID"@en ;
+   samm:description "A unique quality task identifier where these list of parts analysis belong to."@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "BPN-811_2022_000001" .
```
[the remark this answers is not in the document]
Comment (JulianSchererZF) [reply io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl:42]: Currently there is no definition of the ID type which can be used here. The other models from quality also just define this id as bamm/samm-c:Text without any regex

[the remark this answers is not in the document]
Comment (bs-jokri) [reply io.catenax.quality_task_attachment/1.0.0/QualityTaskAttachment.ttl:42]: Ok

**Pull request merged**